### PR TITLE
Expose log level from logging component.

### PR DESCRIPTION
**Pull request by dplepage-dd**
### What does this PR do?
This PR exposes the log level from the logging component. It reexports seelog.LogLevel as log.Level to avoid having the interface explicitly rely on seelog.

This PR also modifies the trace package and the serverless init command, because both define loggers that need to adhere to the log component specification, so I added GetLogLevel() methods to them.

### Motivation
We need this for componentizing netflow - we use a third-party library that uses a different logging library, and as part of connecting those logs to ours we need to set their logging level to match ours.

### Describe how to test/QA your changes
A unit test is already provided for the single function this adds.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Comment by dplepage-dd** [comment]
This could just use seelog.LogLevel if anyone would prefer that; the downside is that it leaks implementation details from the component, but the upside is that seelog implements things like `LogLevel.String()` that I didn't include here. Let me know if you'd like me to do things that way instead.

**Review by ogaca-dd — APPROVED** [review]

**Review by FlorianVeaux — COMMENTED** [review]
Seems like you may need to add the new `GetLogLevel` to this other implementation of the logger ?
https://github.com/DataDog/datadog-agent/blob/602a33f2fc1c9a99689e4156ce76877752e45485/pkg/trace/log/logger.go#L154

File: pkg/trace/log/buflogger.go [hunk 1]
```diff
@@ -93,3 +95,8 @@ func (b *buflogger) Criticalf(format string, params ...interface{}) error {
 
 // Flush implements Logger.
 func (b *buflogger) Flush() {}
+
+// GetLogLevel implements Logger.
+func (b *buflogger) GetLogLevel() (log.Level, error) {
+	return log.DebugLvl, nil
```
Comment (ahmed-mez) [inline pkg/trace/log/buflogger.go:101]: Why debug? Shouldn't it return the actual log level?

Comment (AlexandreYang) [reply pkg/trace/log/buflogger.go:101]: Looks like the `buflogger` is only used for testing.

File: comp/core/log/logger.go [hunk 2]
```diff
@@ -139,6 +157,12 @@ func (*logger) Criticalf(format string, params ...interface{}) error {
 	return log.CriticalStackDepth(2, fmt.Sprintf(format, params...))
 }
 
+// GetLogLevel implements Component#GetLogLevel.
+func (*logger) GetLogLevel() (Level, error) {
```
Comment (AlexandreYang) [inline comp/core/log/logger.go:161]: Shall we test `GetLogLevel()` for each level, the risk is low, but testing that is cheap also.

Not blocker, but I think it's good to have higher coverage.

**Review by AlexandreYang — APPROVED** [review]

File: pkg/trace/log/buflogger.go (hunk 1 above)
Comment (dplepage-dd) [reply pkg/trace/log/buflogger.go:101]: The buflogger doesn't have a log level, it just always logs everything.

Comment (ahmed-mez) [reply pkg/trace/log/buflogger.go:101]: Got it, let's add a code comment then explaining why it always returns Debug.

**Comment by dplepage-dd** [comment]
Closing this - ongoing work to upgrade goflow2 means we soon won't need this feature anymore.

**Pull request closed without merging**